src/Cuco.API/Controllers/UserController.cs

```diff
@@ -0,0 +1,101 @@
+using Cuco.Application.Providers;
+    using Cuco.Commons.Base;
+using Cuco.Domain.Roles.Models.Consts;
+using Cuco.Domain.Roles.Services.Repositories;
+using Cuco.Domain.Users.Models.DTO;
+using Cuco.Domain.Users.Models.Entities;
+using Cuco.Domain.Users.Services.Repositories;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Cuco.API.Controllers;
+
+[ApiController]
+[Produces("application/json")]
+[Route("api/user")]
+public class UserController : ControllerBase
+{
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly IUserProvider _userProvider;
+    private readonly IUserRepository _userRepository;
+
+    public UserController(IUserProvider userProvider, IUserRepository userRepository, IUnitOfWork unitOfWork)
+    {
+        _userProvider = userProvider;
+        _userRepository = userRepository;
+        _unitOfWork = unitOfWork;
+    }
+
+    [HttpPost]
+    [Authorize(Roles = $"{RoleNames.Admin}")]
+    [ProducesResponseType(typeof(UserDto), StatusCodes.Status200OK)]
+    public async Task<ActionResult> AddAsync(
+        [FromServices] IRoleRepository roleRepository,
+        [FromBody] SignUpDto input)
+    {
+        try
+        {
+            var role = await roleRepository.GetAsNoTrackingAsync(input.Role.GetHashCode());
+            var user = new User(input.Name, input.Password, role);
+            _userRepository.Insert(user);
+            if (!_unitOfWork.Commit())
+                return StatusCode(StatusCodes.Status500InternalServerError);
+            var result = new UserDto { Name = user.Name, Role = user.Role };
+            return Ok(result);
+        }
+        catch (Exception e)
```

Comment:
this "try/catch => exception" behaviour seems default in your controllers, and can be moved to an exception handler to be added on your Program.cs as part of the mvc action pipeline. so you don't need to repeat on every action, and you can add special treatments to different things in one place if Its about generating results from exceptions.